### ISSUE TITLE
Unexpected results

### ISSUE THREAD

**Issue by xiankgx** [jingyuanli001/RFR-Inpainting#33]
Dear @jingyuanli001 , I tried to train a model using the places2 dataset. After I have trained for approximately 750,000 iterations (batch size 9 using 3 GPUs; size 384), then I extracted the checkpoint and test it on the coco dataset. I'm getting unsatisfactory results. Why do you think there are obvious "lime water" effects around the masked regions?

The testing masks and the training masks are the same.

![comp_0](https://user-images.githubusercontent.com/4113258/103252597-7aeb8e80-49b8-11eb-9671-d4baeb4dd655.jpg)
![fake_0](https://user-images.githubusercontent.com/4113258/103252601-8048d900-49b8-11eb-9921-05d883f4def3.jpg)
![masked_0](https://user-images.githubusercontent.com/4113258/103252608-8474f680-49b8-11eb-9a21-2242a52e350f.jpg)
![comp_1](https://user-images.githubusercontent.com/4113258/103252630-8dfe5e80-49b8-11eb-8ae1-e8b014a8003b.jpg)
![fake_1](https://user-images.githubusercontent.com/4113258/103252633-8f2f8b80-49b8-11eb-9be5-0d7fe71c7bf2.jpg)
![masked_1](https://user-images.githubusercontent.com/4113258/103252636-8fc82200-49b8-11eb-9e82-f314a2c1127e.jpg)
![comp_2](https://user-images.githubusercontent.com/4113258/103252663-ac645a00-49b8-11eb-8fb2-3820f667cb48.jpg)
![fake_2](https://user-images.githubusercontent.com/4113258/103252664-ad958700-49b8-11eb-9dad-efd8313cac4d.jpg)
![masked_2](https://user-images.githubusercontent.com/4113258/103252665-ae2e1d80-49b8-11eb-9dc6-18c2b566bc63.jpg)
![comp_3](https://user-images.githubusercontent.com/4113258/103252695-d6b61780-49b8-11eb-85d5-d55ce64e5b25.jpg)
![fake_3](https://user-images.githubusercontent.com/4113258/103252696-d7e74480-49b8-11eb-91a1-09fa024d6049.jpg)
![masked_3](https://user-images.githubusercontent.com/4113258/103252698-d87fdb00-49b8-11eb-9e27-5190d66fb3b1.jpg)
![comp_4](https://user-images.githubusercontent.com/4113258/103252702-dc136200-49b8-11eb-9ab2-38cf4fb44461.jpg)
![fake_4](https://user-images.githubusercontent.com/4113258/103252703-dd448f00-49b8-11eb-804f-697bbc578d22.jpg)
![masked_4](https://user-images.githubusercontent.com/4113258/103252704-dddd2580-49b8-11eb-9784-b814eae2b089.jpg)

**Comment by jingyuanli001** [comment]
Hi, have you fine-tuned the model following the procedure in the readme file? The distorted color is very likely to come from inadequate fine-tuning. Fine-tuning his **very very important** here because during fine-tuning, we stopped updating the parameters for batch-normalization which is not stable due to the unpredictable mask region.

**Comment by xiankgx** [comment]
I have not started the finetuning stage yet. Ok, I will follow your instructions to finetune soon. Thank you so much for your swift response and help.

**Comment by xiankgx** [comment]
Dear @jingyuanli001 ,

This is what I got when saving the batch every 200 iterations during training. So, training results looks good, but testing results is totally different. So, you could be right in the batchnorm parameters being unstable. I will start finetune soon and update you.

![input](https://user-images.githubusercontent.com/4113258/103253354-c18eb800-49bb-11eb-909e-6360e863f456.png)
![comp](https://user-images.githubusercontent.com/4113258/103253356-c6536c00-49bb-11eb-849f-4d2d9c9949d0.png)

**Comment by jingyuanli001** [comment]
> Dear @jingyuanli001 ,
> 
> This is what I got when saving the batch every 200 iterations during training. So, training results looks good, but testing results is totally different. So, you could be right in the batchnorm parameters being unstable. I will start finetune soon and update you.
> 
> ![input](https://user-images.githubusercontent.com/4113258/103253354-c18eb800-49bb-11eb-909e-6360e863f456.png)
> ![comp](https://user-images.githubusercontent.com/4113258/103253356-c6536c00-49bb-11eb-849f-4d2d9c9949d0.png)

Yes, this is what we also observed when we were developing our method. During training, the statistic used to normalize the feature map is calculated dynamically based on the instances in the batch, and then it can be considered to be (almost) accurate. However, during testing, the statistic used for normalization is the one recorded during training, and is not calculated from the testing images, and thus it is inaccurate. This then leads to the bad results. So the fine-tuning procedure is actually asking the model to "get familiar with" the fixed statistics that will be used for testing. In fact, I personally consider the batch normalization scheme here to be suboptimal and there should have some better normalization schemes.

**Comment by xiankgx** [comment]
Dear @jingyuanli001 , you were obsolutely right! Fixing the batchnorm parameters helped solved even with just 10k iterations. The following are testing samples at 780,000 iterations. Thank you so much for your help, and good job in the paper. I really like the idea of working inwards from the border with partial convolution and recurrence.

![comp_0](https://user-images.githubusercontent.com/4113258/103271415-2e727400-49f5-11eb-8448-e126e57faa5c.jpg)
![fake_0](https://user-images.githubusercontent.com/4113258/103271418-2fa3a100-49f5-11eb-8ce9-3d585327d335.jpg)
![masked_0](https://user-images.githubusercontent.com/4113258/103271419-303c3780-49f5-11eb-8d89-7e9b1467c0a8.jpg)
![comp_1](https://user-images.githubusercontent.com/4113258/103271425-3500eb80-49f5-11eb-9619-6468db66bef3.jpg)
![fake_1](https://user-images.githubusercontent.com/4113258/103271426-35998200-49f5-11eb-9ef1-35870a296ceb.jpg)
![masked_1](https://user-images.githubusercontent.com/4113258/103271427-36321880-49f5-11eb-92bc-42202ddd1fb4.jpg)
![comp_2](https://user-images.githubusercontent.com/4113258/103271450-41854400-49f5-11eb-91dd-c8064aa79557.jpg)
![fake_2](https://user-images.githubusercontent.com/4113258/103271452-42b67100-49f5-11eb-96e6-5a019880fe03.jpg)
![masked_2](https://user-images.githubusercontent.com/4113258/103271453-434f0780-49f5-11eb-853c-26ddf2167ea4.jpg)

**Comment by 07hyx06** [comment]
Thanks for your code. I download the official celeba weights and directly test the model, but obtain unexpected results (like the model w/o fine-tuning). Did I miss something?
![masked_img_1](https://user-images.githubusercontent.com/52267384/125220904-f334f400-e2f9-11eb-86c6-db291744e8b8.png)
![img_1](https://user-images.githubusercontent.com/52267384/125220910-f5974e00-e2f9-11eb-942c-29921cbce31b.png)

**Comment by htzheng** [comment]
Hi @xiankgx, nice job! Since the official Places2 model is not released yet, I wonder if by any chance you are able to share the Place2 weights so that we could make a fair comparison with RFR. Thank you a lot! :)
